### ISSUE TITLE
DebugMode sample code is unrealistic for real world training use case

### ISSUE THREAD

**Issue by ezyang** [pytorch/pytorch#168973]
### 🐛 Describe the bug

The sample code for DebugMode looks like this:

```
            # Run model first time
            with DebugMode() as debug_mode, DebugMode.log_tensor_hashes():
                model(x)
                logs1 = debug_mode.logs

            # Run again, in exactly the same way
            with DebugMode() as debug_mode, DebugMode.log_tensor_hashes():
                model(x)
                logs2 = debug_mode.logs

            mismatches = DebugMode.check_hash_mismatches(logs1, logs2)
            for m in mismatches:
                print(f"{m['call']}: hash diff {m['rel_diff']:.2e}")
```

But this is unrealistic in a real world training setting, where it is likely that I am trying to diff two different model configs which can't even run in the same process at the same time.

cc @chauhang @penguinwu @yushangdi @pianpwk 

### Versions

main

**Comment by malfet** [comment]
I remember Alban suggested  that it would be good to have separate debug-model label...
